pkg/controllers/node/ipam.go

```diff
@@ -402,6 +415,13 @@ func (c *ipamController) onBlockUpdated(kvp model.KVPair) {
 		log.WithFields(alloc.fields()).Debug("New IP allocation")
 	}
 
+	// If the block is empty, mark it as such. We'll check if it needs to be
+	// cleaned up in the sync loop.
+	delete(c.emptyBlocks, blockCIDR)
+	if n != "" && len(allocatedHandles) == 0 {
```

Comment:
Should probably skip blocks that have allocations with no handles too.  Otherwise, trying to release them below will fail the `empty()` check on `allocationBlock`